packages/grafana-prometheus/src/datasource.ts

```diff
@@ -925,7 +926,21 @@ export class PrometheusDatasource
 
     // We need a first replace to evaluate variables before applying adhoc filters
     // This is required for an expression like `metric > $VAR` where $VAR is a float to which we must not add adhoc filters
-    const expr = this.templateSrv.replace(target.expr, variables, this.interpolateQueryExpr);
+    const expr = this.templateSrv.replace(
+      target.expr,
+      variables,
+      (value: string | string[] = [], variable: QueryVariableModel | CustomVariableModel) => {
+        if (typeof value === 'string' && target.fromExploreMetrics) {
```

Comment:
`target.fromExploreMetrics` strikes me as a code smell...it feels like the Prometheus DS shouldn't be aware of the various things that it's consumed by, including Explore Metrics.

What is it about being from Explore Metrics that matters here? Perhaps this is just a naming issue.

Reply:
It is indeed a code smell but I couldn't find any other way to detect this case. The way explore metrics handles variables is the problem here. Maybe in the future we'll find a better way to handle them. Currently, unfortunately I don't have any other solution. I am open suggestions.